### PR TITLE
types: use TfTypeError instead of Error

### PR DESCRIPTION
**Pull request by dcousens**

### CODE REVIEW

File: src/types.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ var typeforce = require('typeforce')
 
 function nBuffer (value, n) {
   typeforce(types.Buffer, value)
-  if (value.length !== n) throw new Error('Expected ' + (n * 8) + '-bit Buffer, got ' + (value.length * 8) + '-bit Buffer')
+  if (value.length !== n) throw new typeforce.TfTypeError('Expected ' + (n * 8) + '-bit Buffer, got ' + (value.length * 8) + '-bit Buffer')
```
Comment (dcousens) [inline src/types.js]: I should probably change this to `typeforce.typeError` without the `new`?
Then it won't get confusing with the other type exports (all capitalized).

**Comment by rubensayshi** [comment]
``` js
function TfTypeError (type, value) {
  this.tfError = Error.call(this)
  this.tfType = type
  this.tfValue = value

  var message
  Object.defineProperty(this, 'message', {
    get: function () {
      if (message) return message
      message = tfErrorString(type, value)

      return message
    }
  })
}
```

either need a 3rd argument for `message` or a `TfCustomTypeError` I think

**Comment by dcousens** [comment]
That is the code from https://github.com/dcousens/typeforce/blob/master/index.js#L3-L17,  what's up?

**Comment by rubensayshi** [comment]
Can't specify the message like you are trying

**Comment by dcousens** [comment]
`typeforce.TfTypeError((n * 8) + '-bit Buffer', value.length * 8)` should work

**Comment by dcousens** [comment]
updated

**Comment by rubensayshi** [comment]
"Expected 160-bit Buffer, got Number 256"

but good enough for me

**Comment by dcousens** [comment]
@rubensayshi uh,  `(value.length * 8) + '-bit Buffer'`?

**Comment by rubensayshi** [comment]
already tried; `Expected 160-bit Buffer, got String "256-bit Buffer"`

and it throws, while you already had this test `assert(!types.BigInt(new function NotABigInteger () {}))`

what is it supposed to do? throw or return false?

**Comment by rubensayshi** [comment]
ah both are okay it seems judging by the code in typeforce

**Comment by dcousens** [comment]
Just do `typeforce.TfTypeError((n * 8) + '-bit Buffer', value)` then for now.

Therefore we get 

``` javascript
Expected 160-bit Buffer, got Buffer
```

**Comment by rubensayshi** [comment]
imo just `if (arguments.length === 1 && typeof type === "string") { this.message = type }` in `typeforce.TfTypeError` to fix this

**Comment by rubensayshi** [comment]
I'd actually prefer the `got Number 256` because then it's clearer that the length is the issue?

**Comment by dcousens** [comment]
Yeah lets add it to typeforce,  if its no rush I'm happy to wait for that to all go through,  but I have to go to sleep and wake up in 8 hours if you want to PR it all and I'll merge + publish then?

**Comment by rubensayshi** [comment]
sure, no rush on getting this merged

**Pull request merged**